//  Copyright (c) 2015 SONATA-NFV, 5GTANGO, Paderborn University
// ALL RIGHTS RESERVED.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Neither the name of the SONATA-NFV, 5GTANGO, Paderborn University
// nor the names of its contributors may be used to endorse or promote
// products derived from this software without specific prior written
// permission.
//
// This work has also been performed in the framework of the 5GTANGO project,
// funded by the European Commission under Grant number 761493 through
// the Horizon 2020 and 5G-PPP programmes. The authors would like to
// acknowledge the contributions of their colleagues of the SONATA
// partner consortium (www.5gtango.eu).

syntax = "proto3";

// Definition of the gRPC server
service SmpFsmControl {
    // A simple RPC to test things.
    rpc PingPong(Ping) returns (Pong) {}
  
    // A server-to-client streaming RPC.
    rpc ControlFsm(FsmRegistration) returns (stream FsmAction) {}
}


message Ping {
    string text = 1;
}

message Pong {
    string text = 1;
}

message FsmRegistration {
    string name = 1;
    string si_uuid = 2;
}

message FsmAction {
    string name = 1;
    string target_si_uuid = 2;  // uuid of the service instance the action should match to
    string type = 3;  // e.g. "SET_QUARANTAINE_STATE"
    string value = 4;  // e.g. 0 or 1
    uint32 time_created = 5; // unix timestamp
}